platform/linux-generic/pktio/socket.c

```diff
@@ -468,9 +471,22 @@ static int sock_mmsg_send(pktio_entry_t *pktio_entry, int index ODP_UNUSED,
 	if (odp_unlikely(tx_ts_idx && i >= tx_ts_idx))
 		_odp_pktio_tx_ts_set(pktio_entry);
 
-	odp_ticketlock_unlock(&pkt_sock->tx_lock);
+	if (odp_unlikely(tx_compl_enabled))
+		for (j = 0; j < i; j++) {
```

Comment:
This would simplify a lot with another _odp_pktio_send_tx_compl call: send tx compl or free pkt if no compl needed.

Reply:
Done in v2.